source/extensions/filters/network/http_connection_manager/config.cc

```diff
@@ -350,21 +364,36 @@ void HttpConnectionManagerConfig::createFilterChain(Http::FilterChainFactoryCall
 }
 
 bool HttpConnectionManagerConfig::createUpgradeFilterChain(
-    absl::string_view upgrade_type, Http::FilterChainFactoryCallbacks& callbacks) {
-  auto it = findUpgradeCaseInsensitive(upgrade_filter_factories_, upgrade_type);
-  if (it != upgrade_filter_factories_.end()) {
-    FilterFactoriesList* filters_to_use = nullptr;
-    if (it->second != nullptr) {
-      filters_to_use = it->second.get();
-    } else {
-      filters_to_use = &filter_factories_;
-    }
-    for (const Http::FilterFactoryCb& factory : *filters_to_use) {
-      factory(callbacks);
+    absl::string_view upgrade_type, const Http::FilterChainFactory::UpgradeMap* upgrade_map,
```

Comment:
I wonder if we should pass optional UpgradeMaps instead of relying on pointers all around both here and in the router. Is that too heavy handed?

Reply:
I'm fine being overruled as being overly pointer-happy but absl::optional won't take const references, so we'd actually have to store the upgrade map in the router as an absl::optional in order to return a const reference, and that just feels weird to me.

Reply:
I see, and the map is not _really_ optional, it might just be empty if there are no upgrade overrides in the route, and I was seeing it as being nullopt if there were no overrides. Happy to get the opinion of the subsequent reviewer. I don't feel too strongly either way

Reply:
Pointer seems fine to me, but I do have a nit around variable naming. Can we call it `per_route_upgrade_map` or `override_upgrade_map`, etc. to make it clear that these are the override settings?